slowapi/middleware.py

```diff
@@ -51,6 +51,7 @@ async def dispatch(
                 return exception_handler(request, e)
             # request.state._rate_limiting_complete = True
             response = await call_next(request)
-            response = limiter._inject_headers(response, request.state.view_rate_limit)
+            if limiter.enabled:
```

Comment:
The added test will fail without this conditional header injection.

Reply:
Thanks for the contribution and good spot - but maybe we can put:
```python
if not limiter.enabled:
    return await call_next(request)
```
up on line 22 instead so we avoid any unnecessary work being done.

Reply:
gonna approve this anyway - will leave up to you whether to implement the above or keep as is.

Reply:
I think that's a good call - I'll update that.